crates/pallet-domains/src/block_tree.rs

```diff
@@ -267,6 +282,11 @@ pub(crate) fn process_execution_receipt<T: Config>(
                 }
                 let _ = ExecutionInbox::<T>::clear_prefix((domain_id, to_prune), u32::MAX, None);
 
+                ConsensusBlockHash::<T>::remove(
```

Comment:
why not prune it when we see the first ER instead of waiting until challenge period is over since we do not require this for the same ERs which comes from other operators ?

Reply:
Well the ER that is still in the challenge period can be pruned by fraud proof and resubmitted by other operators thus we need the consensus hash for the verification.